### DISCUSSION TITLE
CLI - Pulling environment variables for non-development environment

### DISCUSSION THREAD

**Discussion by bran-dy-b** [vercel/community#545] [Help] [answered]
Is it possible to `vercel pull` environment variables for the preview environment, for example? If not, it makes using Vercel for secret management a bit harder to think about since we need to also deploy/update our mobile app using the same environment variables. Would love to hear any alternative suggestions or solutions as well.

For context, we are trying to avoid having to duplicate all of the environment variables that are in Vercel in GitHub secrets by using the CLI to pull the `.env` file in a GitHub Action. We are doing an OTA update to an expo app in the GitHub Action.

Looking at the docs and googling, I wasn't able to find any solutions.

Appreciate the help/input!

**Comment by leerob** [comment, marked as answer]
Hey! We just merged a PR for this, which you'll be able to use soon: https://github.com/vercel/vercel/pull/7624

Open to any feedback here 😄

**Reply by bran-dy-b** [reply]
Amazing, looking forward to using it :)

Thanks for the quick response!
